### PR TITLE
Added composite repository

### PR DESCRIPTION
**Pull request by dantleech**
This PR introduces a new `CompositeRepository`, as the original Puli `CompositeRepository` has been removed from the `1.0` release.

This version differs from the original:
- Mounted repositories can be seen as children of the root resource.
- Repositories can be mounted within other repositories.
- Repositories can be mounted directly at "/"

For the case where a repository cannot be resolved an `InMemoryRepository` is created containing references to the root resource of each mount point.

### CODE REVIEW

File: Repository/CompositeRepository.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+     *
+     * The path should be either "/" or an immediate child of "/" or
+     * an immediate child of a resource in an existing mounted repository.
+     *
+     * @param string $path
+     * @param ResourceRepository $repository
+     */
+    public function mount($path, ResourceRepository $repository)
+    {
+        $this->repos[$path] = $repository;
```
Comment (dantleech) [inline Repository/CompositeRepository.php:48]: Add validation for existing repositories

File: Repository/CompositeRepository.php [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function get($path)
+    {
+        list($repository, $repoPath, $path) = $this->getRepository($path);
```
Comment (dantleech) [inline Repository/CompositeRepository.php:56]: It occured to me that we could create a wrapped repository which is aware that it is mounted and contains a reference to the mount point. This would illiminate the need for returning a tuple. But, probably not worth the effort.

File: Repository/CompositeRepository.php [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+namespace Symfony\Cmf\Component\Resource\Repository;
+
+use InvalidArgumentException;
+use Puli\Repository\Api\ResourceCollection;
+use Puli\Repository\Api\ResourceNotFoundException;
+use Puli\Repository\Api\ResourceRepository;
+use Puli\Repository\Resource\Collection\ArrayResourceCollection;
+use Webmozart\PathUtil\Path;
+use Webmozart\Assert\Assert;
+use Puli\Repository\InMemoryRepository;
```
Comment (dantleech) [inline Repository/CompositeRepository.php:12]: Clear up unused use statements

File: Repository/CompositeRepository.php [hunk 4]
```diff
@@ -0,0 +1,180 @@
+<?php
+
```
Comment (dantleech) [inline Repository/CompositeRepository.php:2]: Header

**Comment by dantleech** [comment]
@WouterJ the tests are green for the ResourceBundle with this branch now.

**Comment by dantleech** [comment]
btw this implementation more than likely has a few edge cases regarding paths.

**Comment by wouterj** [comment]
@dantleech what do you think: Shall we skip this for the 1.0 release and include it in 1.1? Now we're using Puli-light, resource packages are mainly meant to get admin interface working correctly. So I guess it isn't very important to include this PR?

**Comment by ElectricMaxxx** [comment]
pls reopen if still interessted.
